### ISSUE TITLE
CSRF deprecation, upgrade docs do not state how to upgrade CSRF service

### ISSUE THREAD

**Issue by reecefowell** [symfony/symfony#14899]
The dev bar outputs the following in the tab for "logs".

DEPRECATED - The Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfProviderAdapter class is deprecated since version 2.4 and will be removed in version 3.0. Use the Symfony\Component\Security\Csrf\CsrfTokenManager class instead.

It is not documented anywhere I can see how to deal with this. I upgraded a project from 2.5 to 2.7 and now I get a lot of these errors. All of which I was able to solve, but the CSRF token issue is a mystery.

The default configuration has this
```
        <service id="form.csrf_provider" class="Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfTokenManagerAdapter">
            <argument type="service" id="security.csrf.token_manager" />
        </service>
```

But the file Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfTokenManagerAdapter states
```
    public function getTokenManager($triggerDeprecationError = true)
    {
        if ($triggerDeprecationError) {
            trigger_error('The '.__METHOD__.' method is deprecated since version 2.4 and will be removed in version 3.0. Use the Symfony\Component\Security\Csrf\CsrfTokenManager class instead.', E_USER_DEPRECATED);
        }

        return $this->tokenManager;
    }

    /**
     * {@inheritdoc}
     */
    public function generateCsrfToken($intention)
    {
        trigger_error('The '.__METHOD__.' method is deprecated since version 2.4 and will be removed in version 3.0. Use the Symfony\Component\Security\Csrf\CsrfTokenManager class instead.', E_USER_DEPRECATED);

        return $this->tokenManager->getToken($intention)->getValue();
    }
```

Only things I can find online are suggestions to disable warnings in my php.ini etc. Considering I solved the other issues surely there is a way to appropriate fix this issue while upgrading also?

These warnings are incredibly invasive BTW, not good for DX.

![https://dl.dropboxusercontent.com/u/18267237/Screenshots/Screen%20Shot%202015-06-07%20at%2005.52.41.png](https://dl.dropboxusercontent.com/u/18267237/Screenshots/Screen%20Shot%202015-06-07%20at%2005.52.41.png)

**Comment by jakzal** [comment]
re #14307

**Comment by jakzal** [comment]
@reecefowell do you use the `form.csrf_provider` service directly to access the token manager, or are you getting the deprecation warning with a standard Symfony configuration?

**Comment by reecefowell** [comment]
Standard symfony configuration.

I checked the source, and in the service XML config file found in FrameworkBundle it is using the very class that its shouting deprecation about.

The common consensus seems to be to just add ~E_DEPRECATED to ones php.ini, which doesn't really address the issue.

A common theme it would seem is that most people who have discussed this are not preparing for 3.0 because there is a lack of documentation explaining how to do so. Most of the discussion seems to suggest most people are opting to mute ~E_DEPRECATED instead of dealing with the problem.

Without muting it, the browser debug bar is the least of the worries, any attempt to use app/console results in thousands of lines deprecation warnings so long it buries any meaningful output.

**Comment by jakzal** [comment]
@reecefowell using a deprecated service by framework internally should not trigger a deprecation warning.

**Comment by tkuska** [comment]
i had the same problem, the reason i was using form.csrf_provider in security.yml instead of security.csrf.token_manager

**Comment by jakzal** [comment]
@tkuska this is now documented, see ##14902

**Comment by giosh94mhz** [comment]
This is also triggered by `FOSUserBundle` and will be fixed in release `2.0.0`
